### PR TITLE
Embed the commit hash in the generated assembly

### PR DESCRIPTION
**Pull request by victorhurdugaci**
cc @davidfowl @pranavkm

### CODE REVIEW

File: build/shade/_k-standard-goals.shade [hunk 1]
```diff
@@ -97,8 +97,35 @@ default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
...
+  
     var projectFiles = Files.Include("src/*/project.json").ToList();
-    projectFiles.ForEach(projectFile => DotnetPack(projectFile, BUILD_DIR, Configuration));
+    projectFiles.ForEach(projectFile => 
+    {
+        if (AddAssemblyInfo)
+        {
+            var projectFolder = Path.GetDirectoryName(projectFile);
+            var assemblyInfoFile = Path.Combine(projectFolder, "Properties", "AssemblyInfo.cs");
+            if (File.Exists(assemblyInfoFile))
```
Comment (Eilon) [inline build/shade/_k-standard-goals.shade]: Why is this check necessary? Shouldn't the SHA always be added?

File: build/shade/_k-standard-goals.shade [hunk 2]
```diff
@@ -97,8 +97,35 @@ default PACKAGELIST_JSON_FILENAME = 'NuGetPackageVerifier.json'
...
+            {
+                var commitHashAttribute = String.Format("[assembly: System.Reflection.AssemblyMetadata(\"CommitHash\", \"{0}\")]", commitHash);
+            
+                var buildInfoFile = Path.Combine(projectFolder, "Properties", "BuildInfo.g.cs");
```
Comment (Eilon) [inline build/shade/_k-standard-goals.shade]: Use `.generated.cs` extension?

File: build/shade/_k-standard-goals.shade (hunk 1 above)
Comment (davidfowl) [inline build/shade/_k-standard-goals.shade]: +1 I don't understand what the check is for.

Comment (muratg) [inline build/shade/_k-standard-goals.shade]: Do we want to add it for local builds as well?  Let's say I'm building logging... if I haven't made a check in, but trying the build only, I'll get incorrect SHA in there...

File: build/shade/_k-standard-goals.shade (hunk 2 above)
Comment (muratg) [inline build/shade/_k-standard-goals.shade]: +1... that seems to be the standard.

File: build/shade/_k-standard-goals.shade (hunk 1 above)
Comment (Eilon) [inline build/shade/_k-standard-goals.shade]: Not super concerned about local builds having a SHA. I'd rather do "less" in local builds to keep them fast, if possible (that is, avoid running Git).

Comment (muratg) [inline build/shade/_k-standard-goals.shade]: So keeping this as an "opt-in" with the environment variable is better? (and not changing the local builds)

Comment (victorhurdugaci) [inline build/shade/_k-standard-goals.shade]: Yes, not requiring git on the PATH locally is better, faster, and more reliable. Plus, locally you might have got the code through the git zip file.

**Comment by victorhurdugaci** [comment]
Updated

**Comment by Eilon** [comment]
:shipit:

**Pull request merged**